src/NuGet.Server.V2/Controllers/NuGetODataController.cs

```diff
@@ -0,0 +1,315 @@
+﻿using NuGet.Server.Core.DataServices;
+using NuGet.Server.Core.Infrastructure;
+using NuGet.Server.V2.OData;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.Mime;
+using System.Threading.Tasks;
+using System.Web.Http;
+using System.Web.Http.OData;
+using System.Web.Http.OData.Query;
+
+namespace NuGet.Server.V2.Controllers
+{
+    [NuGetODataControllerConfiguration]
+    public abstract class NuGetODataController : ODataController
```

Comment:
For this one, can you look at the NuGetGallery ODataController / V2 controller? Ideally these use the same techniques (e.g. no `EnableQuery`, but use the conventions + paged query result from there). This will also let us replicate unit tests from the gallery.